### ISSUE TITLE
jobs: mixed-version testing reports unable to find job_info

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#100609]
Found here https://github.com/cockroachdb/cockroach/pull/99866#issuecomment-1496274627
Despite https://github.com/cockroachdb/cockroach/commit/858d874596a0834262eb44a94a4396b531b9ad54  there is a job_info issue:

```
=== RUN   TestCCLLogic_partitioning_enum
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/0817673d1f07fbf06f1393f987f3a3c7/logTestCCLLogic_partitioning_enum31111442
    test_log_scope.go:79: use -show-logs to present logs inline
    testcluster.go:391: migration-manager-find-jobs: system-jobs-scan: relation "system.job_info" does not exist
```

Jira issue: CRDB-26520

**Comment by rafiss** [comment]
Here is a repro that doesn't involve checking out my PR:

Apply this diff:
```
diff --git a/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum b/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum
index 26f4e6a29a2..6b6089b6a17 100644
--- a/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum
+++ b/pkg/ccl/logictestccl/testdata/logic_test/partitioning_enum
@@ -1,3 +1,4 @@
+# LogicTest: default-configs local-mixed-22.2-23.1

 # Enums are supported for partitioning both by list and range.
 # In 20.2 alphas, partitioning by enums was permitted but led to
 ```

Run `./dev testlogic ccl --files=partitioning_enum`.

It doesn't fail every time. But with `./dev testlogic ccl --files=partitioning_enum --count 5` I usually get a failure.

**Comment by rafiss** [comment]
These are the settings used by the `local-mixed-22.2-23.1`: https://github.com/cockroachdb/cockroach/blob/44f3f31bfc98d840576c746a4c3b5c1cd56d5f32/pkg/sql/logictest/logictestbase/logictestbase.go#L447-L454

And this is how the versions are overridden: https://github.com/cockroachdb/cockroach/blob/44f3f31bfc98d840576c746a4c3b5c1cd56d5f32/pkg/sql/logictest/logic.go#L1382-L1394

**Comment by knz** [comment]
@rafiss I don't see this test using `MakeTestingClusterSettingsWithVersions`. Is that needed?

**Comment by rafiss** [comment]
I think @yuzefovich's change in 6ef18ed3e1604435619aa0651e3d666642949e54 as he explained in https://github.com/cockroachdb/cockroach/pull/99987#issuecomment-1496687935 fixes the issue for `local-mixed-22.2-23.1` logic tests.
> it turns out that for tenant configs we need to use `MakeTestingClusterSetting` but for non-tenant configs `MakeClusterSettings`

**Comment by knz** [comment]
> I think @yuzefovich's change in https://github.com/cockroachdb/cockroach/commit/6ef18ed3e1604435619aa0651e3d666642949e54 as he explained in https://github.com/cockroachdb/cockroach/pull/99987#issuecomment-1496687935 fixes the issue for local-mixed-22.2-23.1 logic tests.

:bulb:   ok now that explains a lot of the weirdness. I'll look at the other failing tests to check what needs to be done there.

> These interfaces are pretty hard to use correctly.

Agreed! Adding an assertion in #100684 to clarify.
